src/app/create-event/page.tsx

```diff
@@ -0,0 +1,159 @@
+'use client';
+
+import { NextPage } from 'next';
+import { format } from 'date-fns';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import { Input } from '../../components/Input';
+import { EventTextarea } from './EventTextarea';
+import { postEvent } from '../../api/events';
+import { eventSchema } from './schema';
+
+const CreateEventPage: NextPage = () => {
+  const {
+    register,
+    watch,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(eventSchema),
+  });
+
+  const onSubmit = async (data: any) => {
+    const dateTimeIso = format(new Date(data.date), "yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+    const eventDataFormatted = {
+      ...data,
+      date: dateTimeIso,
+    };
+
+    postEvent(eventDataFormatted);
+  };
+
+  const levels = [
+    { value: 'Beginner', label: 'Začátečník' },
+    { value: 'Advanced', label: 'Pokročilý' },
+    { value: 'Expert', label: 'Expert' },
+    { value: 'Any', label: 'Pro každého' },
+  ];
+
+  const watchedName = watch('name', '');
+  const watchedDescription = watch('description', '');
+  const watchedSport = watch('sport', '');
+  const watchedDate = watch('date');
+  const watchedLocation = watch('location', '');
+  const watchedLevel = watch('level', 'Any');
+  const watchedPrice = watch('price', 0);
+
+  return (
+    <div className="flex flex-col items-center justify-center lg:justify-between xl:flex-row xl:items-start">
+      <h1 className="mt-24 px-20 text-center text-2xl font-medium leading-normal md:mt-14 md:px-0 xl:text-start xl:text-4xl">
+        Vytvořit sportovní událost
+      </h1>
+      <article className="relative mx-5 mr-0 mt-10 flex w-full max-w-xl flex-col rounded-md bg-white text-center shadow-md lg:text-start xl:mr-28 xl:mt-14 xl:px-0">
+        <form
+          onSubmit={handleSubmit(onSubmit)}
+          className="flex flex-col space-y-8 rounded-md bg-white p-7"
+        >
+          <Input
+            register={register}
+            type="name"
+            name="name"
+            label="Název události"
+            placeholder="Název události"
+            errors={errors}
+            watchedValue={watchedName}
+            isEventInput
+          />
+          <EventTextarea
+            register={register}
+            type="text"
+            name="description"
+            label="Popis"
+            placeholder="Popis"
+            errors={errors}
+            watchedValue={watchedDescription}
+          />
+          <Input
+            register={register}
+            type="text"
+            name="sport"
+            label="Sport"
+            placeholder="Sport"
+            errors={errors}
+            watchedValue={watchedSport}
+            isEventInput
+          />
+          <Input
```

Comment:
I recommend renaming this to something less generic - CustomInput, TextField. Or something. Trust me, you will understand when you have bzilion of dependencies that all have `Input` of some sort. Not only the component, but for example Params -> input into a function